src/CalculatorUITestFramework/WindowsDriverLocalService.cs

```diff
@@ -146,20 +147,21 @@ private bool Ping()
         {
             Uri status;
             Uri service = this.ServiceUrl;
-            HttpClient httpClient = new HttpClient();
-            httpClient.Timeout = this.InitializationTimeout;
-
-            if (service.IsLoopback)
-            {
-                status = new Uri("http://localhost:" + Convert.ToString(this.Port) + "/status");
-            }
-            else
+            using (HttpClient httpClient = new HttpClient())
             {
-                status = new Uri(service + "/status");
+                httpClient.Timeout = this.InitializationTimeout;
+
+                if (service.IsLoopback)
+                {
+                    status = new Uri("http://localhost:" + Convert.ToString(this.Port) + "/status");
+                }
+                else
+                {
+                    status = new Uri(service + "/status");
+                }
+
+                return Task.Run(() => httpClient.GetAsync(status)).Result.IsSuccessStatusCode;
```

Comment:
consider `var httpResponse = Task.Run(() => httpClient.GetAsync(status)).ConfigureAwait(false).GetAwaiter().GetResult();` to avoid self-deadlock?